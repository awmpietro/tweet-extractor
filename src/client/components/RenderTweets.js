import React, { useState } from 'react';
import {
  Row,
  Col,
  Pagination,
  PaginationItem,
  PaginationLink,
} from 'reactstrap';
import TweetBox from './TweetBox';

const RenderTweets = ({ results }) => {
  const [currentPage, setCurrentPage] = useState(0);

  const handlePage = (e, index) => {
    e.preventDefault();
    setCurrentPage(index);
  };

  if (results.length === 0) {
    return null;
  }
  const pageSize = 12;
  const pagesCount = Math.ceil(results.length / pageSize);
  return (
    <>
      <Row>
        {results
          .slice(currentPage * pageSize, (currentPage + 1) * pageSize)
          .map((data, i) => (
            <TweetBox result={data} key={data.id_str} />
          ))}
      </Row>
      <Row className="mt-5">
        <Col md="12" lg="12" xs="12" sm="12">
          <Pagination aria-label="Page navigation example">
            <PaginationItem disabled={currentPage <= 0}>
              <PaginationLink
                onClick={(e) => handlePage(e, currentPage - 1)}
                previous
                href="#"
              />
            </PaginationItem>
            {[...Array(pagesCount)].map((page, i) => (
              <PaginationItem active={i === currentPage} key={i}>
                <PaginationLink onClick={(e) => handlePage(e, i)} href="#">
                  {i + 1}
                </PaginationLink>
              </PaginationItem>
            ))}
            <PaginationItem disabled={currentPage >= pagesCount - 1}>
              <PaginationLink
                onClick={(e) => handlePage(e, currentPage + 1)}
                next
                href="#"
              />
            </PaginationItem>
          </Pagination>
        </Col>
      </Row>
    </>
  );
};

export default RenderTweets;
