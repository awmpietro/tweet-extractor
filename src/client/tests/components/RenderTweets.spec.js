import React from 'react';
import RenderTweets from '../../components/RenderTweets';
import renderer from 'react-test-renderer';

describe('LanguageStats', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<RenderTweets results={[]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});