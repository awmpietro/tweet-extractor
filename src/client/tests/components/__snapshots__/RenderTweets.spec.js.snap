// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageStats snapshot renders 1`] = `null`;
